subsys/bluetooth/services/fast_pair/fp_storage/fp_storage_ak.c

```diff
@@ -393,6 +715,14 @@ int fp_storage_ak_save(const struct fp_account_key *account_key)
 
 	if (account_key_count < ACCOUNT_KEY_CNT) {
 		account_key_count++;
+	} else {
+		ak_overwritten = true;
+	}
+
+	if (IS_ENABLED(CONFIG_BT_FAST_PAIR_STORAGE_AK_BOND) &&
+	    bond->bonded) {
+		/* Procedure finished successfully. Setting conn_ctx to NULL. */
+		bond->conn_ctx = NULL;
```

Comment:
Shouldn't we remove the bond in case it is stored with the RPA address instead of the Identity Address?

Reply:
I think it's not possible to get pairing_complete BT callback with the RPA address

Reply:
Should not be possible. Since we store it in NVM it may be worth to assert that assumption.